relay_test.go

```diff
@@ -857,6 +861,11 @@ func TestRelayStalledConnection(t *testing.T) {
 			Succeeded().End()
 		ts.AssertRelayStats(calls)
 	})
+
+	if unreleasedCount, isEmpty := pool.CheckEmpty(); isEmpty != "" || unreleasedCount > 0 {
+		t.Errorf("Frame pool has %v unreleased frames, errors:\n%v\n",
+			unreleasedCount, isEmpty)
+	}
```

Comment:
Totally new to this code-base but - is there a reason not to do this very check for more tests in this file?

Reply:
It requires using the RecordingFramePool which doesn't actually pool, and is significantly slower + has a lock which ends up serializing the relay. This causes tests to behave differently (less like production), so it's only used in some cases where we're ensuring that we're pooling correctly + not reusing released frames. There's also tests that don't release frames (rely on GC instead) for the error cases where plumbing through the "drop" added more complexity and didn't seem worth it at the time (since some of these error cases shouldn't happen much).

Ideally, we'd run all relay tests with the normal frame pool + with the recording frame pool. The TestServer test helpers make it easy to run the same test multiple times with different options, so this could be worth doing, so there's both prod-like coverage, as well as ensuring that pooling works correctly. If there's still cases where the frame is intentionally not released, there can be another option to indicate skipping the "is empty" check.